base/compiler/methodtable.jl

```diff
@@ -94,12 +94,28 @@ is no least element. In both cases `nothing` is returned.
 function findsup(@nospecialize(sig::Type), table::InternalMethodTable)
     min_valid = RefValue{UInt}(typemin(UInt))
     max_valid = RefValue{UInt}(typemax(UInt))
-    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
-                   sig, table.world, min_valid, max_valid)::Union{MethodMatch, Nothing}
+    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
+                   sig, nothing, table.world, min_valid, max_valid)::Union{MethodMatch, Nothing}
     result === nothing && return nothing
     (result.method, WorldRange(min_valid[], max_valid[]))
 end
 
+function findsup(@nospecialize(sig::Type), table::OverlayMethodTable)
+    min_valid = RefValue{UInt}(typemin(UInt))
+    max_valid = RefValue{UInt}(typemax(UInt))
+    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
+                   sig, table.mt, table.world, min_valid, max_valid)::Union{MethodMatch, Nothing}
+    if result === nothing
+        # fall back to the internal method table
```

Comment:
I'm wondering if we should just write these fallback cases (the one above as well similarly):
```
findsup(sig, InternalMethodTable(table))
```
with
```
InternalMethodTable(table::OverlayMethodTable) = InternalMethodTable(table.world)
```
above. 

Just to make it clear what's going on, rather than copy-pasting the same code in both places.

Reply:
That second call to `findsup` will incur two additional allocations, but I think the refactoring would be preferable here since these code paths are not very performance sensitive.